packages/volto/src/components/manage/Controlpanels/ContentTypeSchema.jsx

```diff
@@ -254,88 +160,84 @@ class ContentTypeSchema extends Component {
+                    onClick={onCancel}
+                  >
+                    <Icon
+                      name={clearSVG}
+                      className="circled"
+                      size="30px"
+                      title={intl.formatMessage(messages.cancel)}
+                    />
+                  </Button>
+                </>
+              }
+            />,
+            document.getElementById('toolbar'),
+          )}
+      </div>
+    );
+  }
 
-      return (
-        <div id="page-controlpanel-schema" className="ui container">
-          <Header disabled>
-            {this.props.intl.formatMessage(messages.title, {
-              id: this.props?.schema?.title || this.props.id,
-            })}
-          </Header>
-          <Form
-            ref={this.form}
-            schema={contentTypeSchema}
-            formData={schemaData}
-            pathname={this.props.pathname}
-            onSubmit={this.onSubmit}
-            onCancel={this.onCancel}
-            hideActions
-          />
-          {this.state.isClient &&
-            createPortal(
-              <Toolbar
-                pathname={this.props.pathname}
-                hideDefaultViewButtons
-                inner={
-                  <>
-                    <Button
-                      id="toolbar-save"
-                      className="save"
-                      aria-label={this.props.intl.formatMessage(messages.save)}
-                      onClick={() => this.form.current.onSubmit()}
-                      disabled={this.props.schemaRequest.put.loading}
-                      loading={this.props.schemaRequest.put.loading}
-                    >
-                      <Icon
-                        name={saveSVG}
-                        className="circled"
-                        size="30px"
-                        title={this.props.intl.formatMessage(messages.save)}
-                      />
-                    </Button>
-                    <Button
-                      className="cancel"
-                      aria-label={this.props.intl.formatMessage(
-                        messages.cancel,
-                      )}
-                      onClick={() => this.onCancel()}
-                    >
-                      <Icon
-                        name={clearSVG}
-                        className="circled"
-                        size="30px"
-                        title={this.props.intl.formatMessage(messages.cancel)}
-                      />
-                    </Button>
-                  </>
-                }
-              />,
-              document.getElementById('toolbar'),
-            )}
-        </div>
-      );
-    }
+  return <div />;
+};
 
-    return <div />;
-  }
-}
+ContentTypeSchema.propTypes = {
+  getSchema: PropTypes.func.isRequired,
+  putSchema: PropTypes.func.isRequired,
+  pathname: PropTypes.string.isRequired,
+  id: PropTypes.string.isRequired,
+};
 
 export default compose(
-  injectIntl,
   connect(
```

Comment:
The functional component should use useSelector to get schema and schemaRequest from the Redux state, and useLocation to get the pathname. Then we don't need to use `connect` here.

Reply:
I'll close this PR to reduce noise in the list of PRs ready for review. You are welcome to ask us to reopen it if you update it with this change.